pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -118,10 +142,46 @@ func (cache *claimInfoCache) hasPodReference(UID types.UID) bool {
 	defer cache.RUnlock()
 
 	for _, claimInfo := range cache.claimInfo {
-		if claimInfo.podUIDs.Has(string(UID)) {
+		if claimInfo.PodUIDs.Has(string(UID)) {
 			return true
 		}
 	}
 
 	return false
 }
+
+func (cache *claimInfoCache) addPodReference(claimName, namespace string, podUID types.UID) {
+	cache.Lock()
+	defer cache.Unlock()
+
+	claimInfo, exist := cache.claimInfo[claimName+namespace]
+	if exist {
+		claimInfo.PodUIDs.Insert(string(podUID))
+	} else {
+		klog.InfoS("failed to get claimInfo with claim name %s and namespace %s", claimName, namespace)
+	}
```

Comment:
Thinking about this again though -- why does thismethod  need to hang off the cache at all? Can't it just hang directly off of the `claimInfo` object itself to avoid this issue entirely?